### PR TITLE
Appointment queries

### PR DESCRIPTION
**Pull request by shawnorrific**

### CODE REVIEW

File: db-management/Appointments/queries/getUpcomingDropoffs.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
+const db = require('../../dbConfig');
+
+const getUpcomingDropoffs = () => {
```
Comment (kstensberg) [inline db-management/Appointments/queries/getUpcomingDropoffs.js]: see https://github.com/kstensberg/bizeebird-old/blob/master/bizeebird/Ui/MainWindow.cs#L94-#L127 for reference. 

the current implementation includes all the appointment birds in a twirl that's collapsed by default

File: db-management/Appointments/queries/getUpcomingDropoffs.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    return new Promise((resolve, reject) => {
+        db.serialize(() => {
+            var query = 'SELECT Customers.CustomerId, Customers.Name, Customers.BoardingRate AS Rate, Birds.Name AS BirdName,' +
+            'Birds.Breed, AppointmentBirds.CageNeeded, Appointments.StartTime AS Date FROM Customers ' +
+            'LEFT JOIN Birds ON Birds.Customer_CustomerId = Customers.CustomerId ' +
+            'LEFT JOIN Appointments ON Appointments.Customer_CustomerId = Customers.CustomerId ' +
+            'LEFT JOIN AppointmentBirds ON AppointmentBirds.Appointment_AppointmentId = Appointments.AppointmentId ' +
+            'WHERE Appointments.StartTime > date("now", "-30 days") AND Appointments.Status = 1 ' +
```
Comment (kstensberg) [inline db-management/Appointments/queries/getUpcomingDropoffs.js]: current implementation does not filter by 30 days.  

where a.StartTime >= DateTime.Today &&
a.Status == AppointmentStatus.Scheduled

File: db-management/Appointments/queries/getUpcomingPickups.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+const getUpcomingDropoffs = () => {
+    return new Promise((resolve, reject) => {
+        db.serialize(() => {
+            var query = 'SELECT Customers.CustomerId, Customers.Name, Customers.BoardingRate AS Rate, Birds.Name AS BirdName,' +
+            'Birds.Breed, AppointmentBirds.GroomingWings AS Wings, AppointmentBirds.GroomingNails AS Nails, ' +
+            'Appointments.Notes AS Notes, Appointments.EndTime AS Date FROM Customers ' +
+            'LEFT JOIN Birds ON Birds.Customer_CustomerId = Customers.CustomerId ' +
+            'LEFT JOIN Appointments ON Appointments.Customer_CustomerId = Customers.CustomerId ' +
+            'LEFT JOIN AppointmentBirds ON AppointmentBirds.Appointment_AppointmentId = Appointments.AppointmentId ' +
+            'WHERE Appointments.EndTime > date("now", "-30 days") AND Appointments.Status = 2 ' +
```
Comment (kstensberg) [inline db-management/Appointments/queries/getUpcomingPickups.js]: current implementation does not filter by 30 days.  filter in current implementation

where a.EndTime >= DateTime.Today && 
                                   a.Status == AppointmentStatus.CheckedIn
                                   orderby a.EndTime ascending

File: db-management/Appointments/queries/getUpcomingPickups.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
+const db = require('../../dbConfig');
+
+const getUpcomingDropoffs = () => {
```
Comment (kstensberg) [inline db-management/Appointments/queries/getUpcomingPickups.js]: see https://github.com/kstensberg/bizeebird-old/blob/master/bizeebird/Ui/MainWindow.cs#L151-#L185 for reference

the current implementation includes all the appointment birds in a twirl that's collapsed by default

File: ipc.js [hunk 5]
```diff
@@ -16,25 +16,6 @@ ipcMain.handle('searchHistory', async function(event, searchString) {
...
-ipcMain.handle('getUpcomingDropoffs', async function() {
-    return [{
-        Date: '08/20/22',
-        Name: 'Kevin Stensberg',
-        BirdName: 'River',
-        BirdBreed: 'Dog',
-        CageNeeded: true
-    }];
-});
+ipcMain.handle('getUpcomingDropoffs', require('./db-management/Appointments/queries/getUpcomingDropoffs'));
```
Comment (kstensberg) [inline ipc.js]: doesn't this now need to pass in a database connection?

**Review by kstensberg — CHANGES_REQUESTED** [review]

File: ipc.js (hunk 5 above)
Comment (shawnorrific) [reply ipc.js]: I ended up undoing that work because I couldn't get it working on tests. Let's work on this tomorrow.

**Pull request merged**